app/models/grouping.rb

```diff
@@ -59,6 +59,7 @@ class Grouping < ApplicationRecord
            -> {
              left_outer_joins(role: :user,
                               test_group_results: [:test_group, :test_results])
+               .where.not('test_groups.id': nil)
```

Comment:
@mishaschwartz sorry for the delay in looking at this one. All of the changes look good, except this one here I think is from an earlier iteration of your changes. I believe we decided to ensure `TestGroupResult`s always had an associated test group (to avoid the "test group id is nil")?

If so, I don't think this `where` is necessary, and moreover we can change `TestGroupResult` association `belongs_to :test_group, optional: true` to remove the `optional: true`, and ideally make a database migration to make the column non-null.

But perhaps I'm not remembering correctly, or missing a use case for keeping this column as nullable?